src/platform/3ds/3ds_audio.cpp

```diff
@@ -67,15 +67,15 @@ void CtrAudio::BGM_Play(std::string const& file, int volume, int pitch, int fade
 	if (!dsp_inited)
 		return;
 
-	FILE* filehandle = FileFinder::fopenUTF8(file, "rb");
-	if (!filehandle) {
+	auto filestream = FileFinder::OpenInputStream(file, std::ios::ios_base::in | std::ios::ios_base::binary);
+	if (!filestream) {
 		Output::Warning("Couldn't play BGM %s: File not readable", file.c_str());
 		return;
 	}
 
 	LockMutex();
-	bgm_decoder = AudioDecoder::Create(filehandle, file);
-	if (bgm_decoder && bgm_decoder->Open(filehandle)) {
+	bgm_decoder = AudioDecoder::Create(filestream, file);
+	if (bgm_decoder && bgm_decoder->Open(filestream)) {
```

Comment:
For the `bgm_decoder` why do we have to pass the stream to the create method and then open it again?

Reply:
strange API design: Create will return an instance of a suitable audio decoder and Open() will do the actual processing. Should be improved but not here (and not before this PR, I don't want to rebase this :P)